src/main/java/com/ita/if103java/ims/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,103 @@
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+@Service
+@PropertySource("classpath:application.properties")
+public class UserServiceImpl implements UserService {
+
+    @Value("${mail.activationURL}")
+    private String activationURL;
+
+    private UserDao userDao;
+    private UserDtoMapper userDtoMapper;
+    private MailService mailService;
+
+    @Autowired
+    public UserServiceImpl(UserDao userDao, UserDtoMapper userDtoMapper, MailService mailService) {
+        this.userDao = userDao;
+        this.userDtoMapper = userDtoMapper;
+        this.mailService = mailService;
+    }
+
+    @Override
+    public UserDto create(UserDto userDto) {
+        User createdUser = userDao.create(userDtoMapper.convertUserDtoToUser(userDto));
+        activationURL += createdUser.getEmailUUID();
+        sendActivationMessage(userDto, activationURL);
+        return userDtoMapper.convertUserToUserDto(createdUser);
+    }
+
+    @Override
+    public UserDto findById(Long id) {
+        return userDtoMapper.convertUserToUserDto(userDao.findById(id));
+    }
+
+    @Override
+    public UserDto findByAccountId(Long accountID) {
+        return userDtoMapper.convertUserToUserDto(userDao.findByAccountId(accountID));
+    }
+
+    @Override
+    public UserDto update(UserDto userDto) {
+        User updatedUser = userDtoMapper.convertUserDtoToUser(userDto);
+        //Activating status can't be changed in this way
+        User DBUser = userDao.findById(updatedUser.getId());
+        updatedUser.setActive(DBUser.isActive());
+        return userDtoMapper.convertUserToUserDto(userDao.update(updatedUser));
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return userDao.softDelete(id);
+    }
+
+    @Override
+    public List<UserDto> findAll() {
+        return userDtoMapper.convertToUserDtoList(userDao.findAll());
+    }
+
+    @Override
+    public UserDto findByEmail(String email) {
+        return userDtoMapper.convertUserToUserDto(userDao.findByEmail(email));
+    }
+
+    @Override
+    public boolean updatePassword(Long id, String newPassword) {
+        return userDao.updatePassword(id, newPassword);
+    }
+
+    @Override
+    public boolean activateUser(String emailUUId) {
+        User activatedUser = userDao.findByEmailUUID(emailUUId);
+        ZonedDateTime createdDateTime = activatedUser.getCreatedDate();
+        ZonedDateTime currrentDateTime = ZonedDateTime.now(ZoneId.systemDefault());
+
+        Duration duration = Duration.between(createdDateTime, currrentDateTime);
+        long diffHours = Math.abs(duration.toHours());
+        if (diffHours <= 24) {
+            activatedUser.setActive(true);
+            userDao.update(activatedUser);
+            return true;
+        }
+        return userDao.hardDelete(activatedUser.getId());
```

Comment:
I am not sure if this is good and how are you going to handle result of this method, but in case of successful hard delete your method will return true as well as after successful activation :)

Reply:
Yes, I agree but for now, I don't know how to make a message which of two actions has been done.

Reply:
I know. I will return false in delete case)